src/infra/raft/v2/RaftCore.cpp

```diff
@@ -909,11 +919,38 @@ void RaftCore::stepDown(uint64_t newTerm) {
 
     /// notify monitor
     mLeadershipGauge.set(0);
-    /// resume election timer
+    mMajorityIndexGauge.set(0);
+
+    for (auto &p : mPeers) {
+      auto &peer = p.second;
+      /// followers match index & lag
+      gringofts::getGauge("match_index", {{"address", peer.mAddress}})
+          .set(0);
+    }
+    gringofts::getGauge("match_index", {{"address", mSelfInfo.mAddress}})
+        .set(0);
     updateElectionTimePoint();
   }
 }
 
+void RaftCore::getInSyncFollowers( 
+    const int64_t &threshold,
+    std::vector<MemberOffsetInfo> *mInSyncFollowers) const{
```

Comment:
If you need to return a vector ordered by mId, why not just return a std::map<id, meminfo> ?

Reply:
that does not make much difference, i think